application/src/test/java/uk/nhs/digital/uec/api/service/PostcodeMappingServiceTest.java

```diff
@@ -47,40 +52,47 @@ public void init() {
   }
 
   @Test
-  public void testGetByPostCodeInAndName() throws InvalidPostcodeException {
+  public void testGetByPostCodeInAndName() throws InvalidPostcodeException, NotFoundException {
     when(postcodeMappingRepository.findByPostCodeAndName("WA11QY", serviceName))
-        .thenReturn(postcodeMappingOptList);
+        .thenReturn(postcodeMappingOptList.get(0));
     List<PostcodeMapping> findByPostCodeIn =
         postcodeMappingService.getByPostCodesAndName(postCodes, serviceName);
     assertFalse(findByPostCodeIn.isEmpty());
   }
 
   @Test
-  public void testGetByPostCodes() throws InvalidPostcodeException {
-    when(postcodeMappingRepository.findByPostCode("WA11QY")).thenReturn(postcodeMappingOptList);
+  public void testGetByPostCodes() throws InvalidPostcodeException, NotFoundException {
+    when(postcodeMappingRepository.findByPostCode("WA11QY"))
```

Comment:
Can we move the postcode into a final variable so the code smells check doesn't pick it up? I think it is used in 3 places. This is my fault as I only noticed this up after I had merged in my changes